### PR TITLE
Fix some dependencies

### PR DESCRIPTION
**Pull request by silverv**
This is not ready for merge yet

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -7,6 +7,6 @@
 [submodule "kucher/libraries/dataclasses"]
 	path = kucher/libraries/dataclasses
 	url = https://github.com/ericvsmith/dataclasses
-[submodule "kucher/libraries/quamash"]
-	path = kucher/libraries/quamash
-	url = https://github.com/harvimt/quamash
+[submodule "kucher/libraries/qasync"]
+	path = kucher/libraries/qasync
+	url = https://github.com/silverv/qasync.git
```
Comment (pavel-kirienko) [inline .gitmodules]: Unless you introduced custom patches, let us use the upstream here.

File: kucher/__init__.py [hunk 2]
```diff
@@ -34,6 +34,7 @@
     os.path.join(THIRDPARTY_PATH_ROOT, 'popcop', 'python'),
     os.path.join(THIRDPARTY_PATH_ROOT, 'construct'),
     os.path.join(THIRDPARTY_PATH_ROOT, 'quamash'),
```
Comment (pavel-kirienko) [inline kucher/__init__.py]:
```suggestion
```

**Review by pavel-kirienko — COMMENTED** [review]
Nice. I see that Construct is still a submodule rather than a pip dependency.

File: .gitmodules (hunk 1 above)
Comment (silverv) [reply .gitmodules]: introduced one change

Comment (pavel-kirienko) [reply .gitmodules]: Okay. See Slack, your patch is not required, please make this point upstream.

**Comment by pavel-kirienko** [comment]
Did you apply Black to all files in this PR? I see a lot of trivial formatting changes across the codebase.

**Comment by silverv** [comment]
> Did you apply Black to all files in this PR? I see a lot of trivial formatting changes across the codebase.

Yes, to the files of Kucher not the libraries

**Comment by pavel-kirienko** [comment]
Runs fine on clean Windows 11

image

File: kucher/__init__.py [hunk 3]
```diff
@@ -27,20 +27,20 @@
...
+    os.path.join(THIRDPARTY_PATH_ROOT, "construct"),
+    os.path.join(THIRDPARTY_PATH_ROOT, "qasync"),
 ]
 
 # 'dataclasses' module is included in Python libraries since version 3.7. For Python versions below, the dataclass
 # module located in the 'libraries' directory will be used. It is not compatible with Python 3.7, so we only declare
 # its path if Python version is below 3.7. Otherwise, the built-in module will be used by default.
 if sys.version_info[:2] < (3, 7):
-    THIRDPARTY_PATH.append(os.path.join(THIRDPARTY_PATH_ROOT, 'dataclasses'))
+    THIRDPARTY_PATH.append(os.path.join(THIRDPARTY_PATH_ROOT, "dataclasses"))
```
Comment (pavel-kirienko) [inline kucher/__init__.py]: This should be deleted

File: .gitmodules [hunk 4]
```diff
@@ -1,12 +1,9 @@
 [submodule "kucher/libraries/popcop"]
 	path = kucher/libraries/popcop
 	url = https://github.com/Zubax/popcop
-[submodule "kucher/libraries/construct"]
-	path = kucher/libraries/construct
-	url = https://github.com/construct/construct
 [submodule "kucher/libraries/dataclasses"]
 	path = kucher/libraries/dataclasses
 	url = https://github.com/ericvsmith/dataclasses
```
Comment (pavel-kirienko) [inline .gitmodules]: This is no longer needed and should be deleted. Dataclasses are now part of the standard library.

File: kucher/__init__.py [hunk 5]
```diff
@@ -17,7 +17,7 @@
 import sys
 
 if sys.version_info[:2] < (3, 6):
-    raise ImportError('A newer version of Python is required')
+    raise ImportError("A newer version of Python is required")
```
Comment (pavel-kirienko) [inline kucher/__init__.py]: I suggest we delete this as python3.10 is used in all modern OSes. Or at least we bump the version number to 3.10.

**Review by pavel-kirienko — CHANGES_REQUESTED** [review]
Intermediate review, will continue later

File: kucher/model/device_model/device_info_view.py [hunk 6]
```diff
@@ -64,128 +70,149 @@ def populate(prototype: NodeInfoMessage):
 class MathRange:
     min: float
     max: float
+    # _io: object
```
Comment (pavel-kirienko) [inline kucher/model/device_model/device_info_view.py]: Please remove commented-out code

File: requirements-dev-windows.txt [hunk 7]
```diff
@@ -7,8 +7,9 @@ setuptools
 wheel
 
 # Test dependencies
-pytest >= 3.3
+pytest >= 7.*
```
Comment (pavel-kirienko) [inline requirements-dev-windows.txt]: Did you mean `==`?

**Review by pavel-kirienko — CHANGES_REQUESTED** [review]
Okay, I reviewed this against your autoformatted branch and it looks okay except for a couple of very minor changes. I reviewed the diff using this:

```
git difftool ca08f8f61f2219a3e15e0be1ff5726418aadf804 9d41eab7be6732bb9b34c2fd65b9b0397e8f99c7
```

Please in the future do not mix autoformatting with any non-trivial changes, ever, for obvious reasons.

File: requirements-dev-windows.txt (hunk 7 above)
Comment (silverv) [reply requirements-dev-windows.txt]: yes

**Comment by pavel-kirienko** [comment]
I see you fixed one occurrence of `# _io: object`, but there are more. Please remove all of them.

**Review by pavel-kirienko — APPROVED** [review]

**Pull request merged**